model/task/db.go

```diff
@@ -888,15 +888,6 @@ func FindUniqueBuildVariantNamesByTask(projectId string, taskName string, repoOr
 		},
 	}
 
-	// sort by most recent version to get the most recent display names for the build variants first
-	sortByOrderNumber := bson.M{
```

Comment:
Not related to 16mb limit but its a performance optimization for a query used on the task history page. 
This sort was preventing the use of an index and was very slow on large projects like `mongodb-mongo-*`.

Its no longer necessary since we are limiting to the 50 most recent versions we don't have to worry about older build variant names being shown.